src/Microsoft.AspNet.Server.Kestrel/Networking/Libuv.cs

```diff
@@ -199,6 +200,9 @@ public void async_init(UvLoopHandle loop, UvAsyncHandle handle, uv_async_cb cb)
         protected Func<UvAsyncHandle, int> _uv_async_send;
         public void async_send(UvAsyncHandle handle)
         {
+            // Can't Assert with .Validate as that checks threadId
+            // and this function is to post to correct thread.
+            Debug.Assert(!handle.IsInvalid, "Handle is invalid");
```

Comment:
If this was intended to prevent AVs, it could only have any affect on debug builds. I don't think even Trace.Assert would save us from the AVs we've been seeing.

This might be nice to have, but I doubt this fixes any bugs.